### PR TITLE
postcss-preset-env : add cascade layers

### PR DESCRIPTION
**Pull request by romainmenke**

### CODE REVIEW

File: plugins/postcss-nesting/test/layer.css [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+.contained {
+	background-color: blue;
+	height: 100px;
+	width: 100px;
+
+	@layer A {
+		/* There is no definition for this behavior */
+		background-color: red;
+	}
```
Comment (romainmenke) [inline plugins/postcss-nesting/test/layer.css:13]: I have no idea if this should be allowed and how it should work exactly.

Mirroring `@media` and `@supports` handling seemed logical and should be ok for now.

_Not handling it here causes issues in the `cascade-layer` plugin if it encounters `@layer` in a style rule._

File: plugins/postcss-cascade-layers/src/index.ts [hunk 2]
```diff
@@ -22,7 +22,7 @@ const creator: PluginCreator<pluginOptions> = (opts?: pluginOptions) => {
 
 	return {
 		postcssPlugin: 'postcss-cascade-layers',
-		Once(root: Container, { result }: { result: Result }) {
+		OnceExit(root: Container, { result }: { result: Result }) {
```
Comment (romainmenke) [inline plugins/postcss-cascade-layers/src/index.ts:25]: Better interop with `:has`, `:is`, `@custom-selector`, nesting, ...

**Comment by romainmenke** [comment]
Size diff with and without some cascade layers : 

with cascade layers :

```
+--------------------------------------------------------------+
| Size         | 33.82 KiB                                     |
|--------------------------------------------------------------|
| Gzipped      | 5.26 KiB                                      |
+--------------------------------------------------------------+
```

without :

```
+--------------------------------------------------------------+
| Size         | 27.07 KiB                                     |
|--------------------------------------------------------------|
| Gzipped      | 5.04 KiB                                      |
+--------------------------------------------------------------+
```

File: plugins/postcss-nesting/test/layer.css (hunk 1 above)
Comment (Antonio-Laguna) [reply plugins/postcss-nesting/test/layer.css:13]: That makes sense, I did a brief integration test and saw the issues!

**Review by Antonio-Laguna — APPROVED** [review]
This LGTM.  I'm kind of nervous with this plugin because of how drastic the transformations are! Thanks for checking this so thoroughly

**Comment by romainmenke** [comment]
updated docs : https://github.com/csstools/postcss-plugins/blob/preset-env-add-cascade-layers--compassionate-giant-panda-bear-bf23b5ebc4/plugins/postcss-cascade-layers/README.md#how-it-works

**Review by Antonio-Laguna — APPROVED** [review]
Great work, love the docs, I think this definitely helps with #376

**Pull request merged**